src/OpenLoco/src/Map/TileManager.cpp

```diff
@@ -478,15 +480,36 @@ namespace OpenLoco::World::TileManager
     }
 
     // 0x00462926
-    bool canConstructAt(const World::Pos2& pos, uint8_t baseZ, uint8_t clearZ, const QuarterTile& qt)
+    static bool canConstructAt(const World::Pos2& pos, uint8_t baseZ, uint8_t clearZ, const QuarterTile& qt, void* clearFunction, uint8_t flags)
     {
+        _F00138 = clearFunction != nullptr ? reinterpret_cast<uint32_t>(clearFunction) : 0xFFFFFFFF;
+        _F00166 = flags;
+
         registers regs;
         regs.ax = pos.x;
         regs.cx = pos.y;
         regs.dl = baseZ;
         regs.dh = clearZ;
         regs.bl = qt.getBaseQuarterOccupied() | (qt.getZQuarterOccupied() << 4);
-        return !(call(0x00462926, regs) & Interop::X86_FLAG_CARRY);
+        return !(call(0x00462937, regs) & Interop::X86_FLAG_CARRY);
+    }
+
+    // 0x00462908
+    bool sub_462908(const World::Pos2& pos, uint8_t baseZ, uint8_t clearZ, const QuarterTile& qt, void* clearFunction)
+    {
+        return canConstructAt(pos, baseZ, clearZ, qt, clearFunction, (1 << 7) | (1 << 0));
+    }
+
+    // 0x00462917
+    bool sub_462917(const World::Pos2& pos, uint8_t baseZ, uint8_t clearZ, const QuarterTile& qt, void* clearFunction)
+    {
+        return canConstructAt(pos, baseZ, clearZ, qt, clearFunction, (1 << 0));
+    }
+
+    // 0x00462926
+    bool canConstructAt(const World::Pos2& pos, uint8_t baseZ, uint8_t clearZ, const QuarterTile& qt, void* clearFunction)
```

Comment:
in OpenRCT2 we called it the equivalent of canConstructAt if it didn't pass a clear function and canConstructWithClearAt if it did include a clear function. `0x00462926` is always one without a clear function.

Reply:
Ahh, I see now. I'll remove the parameter for the outward-facing function, and rename the internal one, then.